### DISCUSSION TITLE
Watermark Pages while retaining bookmarks?

### DISCUSSION THREAD

**Discussion by metro69** [py-pdf/pypdf#716] [Q&A]
When using a simple a simple iteration over a pdf file and adding watermarks using mergePage function, bookmarks are not retained.

The merger class does retain bookmarks, but cannot do watermarking.

Is there an example that shows how to do watermarking while retaining bookmarks?

**Comment by ndunn219** [comment]
I have this same question. I thought there might be a way to do it by looping through the outline items of the pdfreader and writing them to the pdfwriter, but I couldn't figure out how to do that. It'd be awesome if it could be as easy as:

```
pdfwriter.outline = pdfreader.outline
```

**Reply by pubpub-zz** [reply]
`PdfWriter.Append()` copies the outlines of the selected copied pages. Once you have copied the pages, you can then merge the watermark directly on the pages.
